context : remove logits_all flag

ggerganov opened the pull request:
This flag has been deprecated for quite some time and in order to simplify the logic in `llama_context` it's better to remove it. To achieve the same functionality, set `llama_batch::logits` accordingly.

### `libllama` API Changes

- Remove `logits_all` flag
- Move the `bool` flags at the end of the struct

ggerganov committed:
context : remove logits_all flag

ggml-ci

slaren reviewed (commented):
I think it still needs to be removed from `llama_context_params`.

ggerganov committed:
llama : remove logits_all flag + reorder llama_context_params

ggml-ci

ggerganov commented:
Removed the flag from `llama_context_params` and also moved the `bools` to the end of the struct. Will update the API changelog with this breaking change.
